### PR TITLE
Fix issues #12 and #17

### PR DESCRIPTION
**Pull request by hajgato**
Fixes for https://github.com/hpcugent/slurm-torque-wrappers/issues/12 and https://github.com/hpcugent/slurm-torque-wrappers/issues/17

### CODE REVIEW

File: files/qsub.pl [hunk 1]
```diff
@@ -190,7 +189,8 @@ sub make_command
         'p=i'      => \$priority,
         'pe=s{2}'  => \@pe_ev_opts,
         'P=s'      => \$wckey,
-        'q=s'      => \$destination,
+        'q=s'      => sub { fatal("In SLURM no queues defined. " .
```
Comment (boegel) [inline files/qsub.pl]: "should be "no queues are defined."

Also, didn't we plan to make this a warning, and simply ignore whatever is passed to -q`?

File: files/qsub.pl [hunk 2]
```diff
@@ -289,6 +289,9 @@ sub make_command
     my @intcommand;
 
     if ($interactive || $fake) {
+        if ($script) {
+            fatal("Intreactive jobs are not allowed to run a jobscript (in this case: \"$script\")");
```
Comment (boegel) [inline files/qsub.pl]: `Interactive` (typo)

File: files/qsub.pl [hunk 3]
```diff
@@ -526,6 +528,12 @@ sub parse_script
     my @lines;
     @lines = split("\n", $txt) if defined $txt;
 
+    # Do not let users use #PBS -q directive
+    foreach my $line (@lines) {
+        if ($line =~ m/\s*#PBS.*?\s-q/) {
+            fatal("In SLURM no queues defined. Please do not use \"#PBS -q <QUEUE>\" directive in the jobscript");
```
Comment (boegel) [inline files/qsub.pl]: Also here, should be a warning + simply ignore whatever value is passed?

Comment (hajgato) [reply files/qsub.pl]: The problem is this part handled by slurm pbs plugin. So if we issue a warning, the plugn will still pick up the queue.

File: files/qsub.pl (hunk 1 above)
Comment (hajgato) [reply files/qsub.pl]: See my comment at the directives. I prefer to have uniform behavior.

File: files/qsub.pl (hunk 3 above)
Comment (boegel) [reply files/qsub.pl]: Can we strip out the `#PBS -q` line from the job script before it reaching the Slurm PBS plugin?

File: files/qsub.pl (hunk 1 above)
Comment (boegel) [reply files/qsub.pl]: Very much +1 on uniform behavior, I overlooked that the `#PBS` lines in the job script are handled elsewhere... Early exiting with an error is definitely better if we can't control that easily.

File: files/qsub.pl (hunk 3 above)
Comment (hajgato) [reply files/qsub.pl]: Not sure it is a good idea to modify the script of the users. Right now the original script passed to sbatch, so we can still overwritten those values with command line. I cannot create a "no queue" command line option for sbatch. If we ever rewrite the wrapper, then we can do tricks (we give only the rewritten stdin (made from batch script) to slurm, then we can do that),

File: files/qsub.pl (hunk 2 above)
Comment (hajgato) [reply files/qsub.pl]: Will fix the typos in the next PR, which will fix issue 17. (will submit soon)

File: files/qsub.pl (hunk 3 above)
Comment (hajgato) [reply files/qsub.pl]: I am not sure that it is a good idea to made this big change in the current wrapper. (the stdin trick)

Comment (boegel) [reply files/qsub.pl]: Fine, then error out is the next best thing, it's not a huge problem as long as the error message is clear (which it is, except for the silly typos ;-) )

**Pull request merged**